db/external_sst_file_ingestion_job.h

```diff
@@ -196,6 +224,18 @@ class ExternalSstFileIngestionJob {
   // file_checksum_gen_factory is set, DB will generate checksum each file.
   bool need_generate_file_checksum_{true};
   std::shared_ptr<IOTracer> io_tracer_;
+
+  // Below are variables used in (un)registering range for this ingestion job
+  //
+  // FileMetaData used in intputs of compactions equivelant to this ingestion
+  // job
+  std::vector<FileMetaData*> compaction_input_metdatas_;
+  // Map from output level to input of compactions equivelant to this ingestion
+  // job
+  std::map<int, std::vector<CompactionInputFiles>>
+      output_level_to_file_ingesting_compaction_input_;
+  // Compactions equivelant to this ingestion job
+  std::vector<Compaction*> file_ingesting_compactions_;
```

Comment:
[Note] Wish I could have used std::unique_ptr<> for `compaction_input_metdatas_` and `file_ingesting_compactions_` and avoid explicit deletion in ~ExternalSstFileIngestionJob(). But this mysterious window failure  [] stopped me from doing so.